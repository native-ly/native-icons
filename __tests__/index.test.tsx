import React from 'react'
import { render } from '@testing-library/react-native'

import NativeIcon, { IconsProvider, FontType } from '../src'

type TestCase = [icon: string, type: FontType]

const testCases: TestCase[] = [
  ['plus', 'ant-design'],
  ['plus', 'entypo'],
  ['plus', 'evil-icons'],
  ['plus', 'feather'],
  ['plus', 'font-awesome'],
  ['plus', 'font-awesome-5'],
  ['facebook', 'fontisto'],
  ['plus', 'foundation'],
  ['ios-add', 'ionicons'],
  ['add', 'material-icons'],
  ['plus', 'material-community-icons'],
  ['plus', 'octicons'],
  ['facebook', 'zocial'],
  ['plus', 'simple-line-icons'],
]

describe('NativeIcons', () => {
  it.each(testCases)('should render %s from %s', (icon, type) => {
    const { toJSON } = render(<NativeIcon type={type} name={icon} />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render add icon from default type â€“ material-icons', () => {
    const { toJSON } = render(<NativeIcon name="add" />)

    expect(toJSON()).toMatchSnapshot()
  })

  it.each(testCases)(
    'should render %s from %s using value from context',
    (icon, type) => {
      const { toJSON } = render(
        <IconsProvider type={type}>
          <NativeIcon name={icon} />
        </IconsProvider>
      )

      expect(toJSON()).toMatchSnapshot()
    }
  )

  it('should override icon type from context with type from icon', () => {
    const { toJSON } = render(
      <IconsProvider type="ionicons">
        <NativeIcon name="add" type="material-icons" />
        <NativeIcon name="ios-add" />
      </IconsProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render icon with default type', () => {
    const { toJSON } = render(
      <IconsProvider>
        <NativeIcon name="add" />
      </IconsProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
