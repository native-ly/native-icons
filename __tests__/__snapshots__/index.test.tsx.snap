// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeIcons should render add from material-icons 1`] = `
<Text
  allowFontScaling={false}
  icon="add"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render add icon from default library â€“ material-icons 1`] = `
<Text
  allowFontScaling={false}
  icon="add"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render facebook from fontisto 1`] = `
<Text
  allowFontScaling={false}
  icon="facebook"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Fontisto",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render facebook from zocial 1`] = `
<Text
  allowFontScaling={false}
  icon="facebook"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "zocial",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render ios-add from ionicons 1`] = `
<Text
  allowFontScaling={false}
  icon="ios-add"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from ant-design 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "anticon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from entypo 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Entypo",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from evil-icons 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "EvilIcons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from feather 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Feather",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from font-awesome 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from font-awesome-5 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome5Free-Regular",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "400",
      },
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from foundation 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "fontcustom",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from material-community-icons 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from octicons 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Octicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NativeIcons should render plus from simple-line-icons 1`] = `
<Text
  allowFontScaling={false}
  icon="plus"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "simple-line-icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
